### PR TITLE
Furo theme

### PR DESCRIPTION
**Pull request by dwelsch-esi**
Set up beta doc site using furo theme.
Includes:
- Fill out folder structure for all TOC sections. Complete AdaRound example.
- Make beta doc build optional for agile development.
- Swap in furo theme for look & feel and right-side menu.
- Refine furo theme options.
- Set up index for API ref.

### CODE REVIEW

File: Docs/beta/snippets/torch/apply_adaround.py [hunk 1]
```diff
@@ -1,40 +1,3 @@
...
-#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
-#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
-#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-#  POSSIBILITY OF SUCH DAMAGE.
-#
-#  SPDX-License-Identifier: BSD-3-Clause
-#
-#  @@-COPYRIGHT-END-@@
-# =============================================================================
-# pylint: disable=all
```
Comment (quic-kyunggeu) [inline Docs/beta/snippets/torch/apply_adaround.py]: Unfortunately our Github [repolinter](https://github.com/quic/aimet/actions/runs/11860256600/job/33055087683) complains if any .py file doesn't include this copyright header😢 
image

Could you please restore this header?

Comment (quic-kyunggeu) [reply Docs/beta/snippets/torch/apply_adaround.py]: (I'd like to loosen the criteria if possible, but we have to first consult @quic-akhobare and others to do so)

File: Docs/beta/_templates/ethical-ads.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{# Add the ethical ads div, when generating documentation on Read the Docs #}
```
Comment (quic-akhobare) [inline Docs/beta/_templates/ethical-ads.html]: What does this do?

Comment (dwelsch-esi) [reply Docs/beta/_templates/ethical-ads.html]: TL;DR: We can delete it.

It's a component for the left sidebar that supports ReadTheDocs' advertising model. We don't use it.

All the flies in the _templates directory are already in the theme's package; they're not needed here unless we modify them. The only ones we're likely to modify are brand.html and navigation.html, which are the title heading and the TOC for the left sidebar.

File: Docs/beta/featureguide/adaround.rst [hunk 3]
```diff
@@ -152,40 +168,53 @@ Next steps
...
+.. tab-set::
+    :sync-group: platform
 
-    .. tab:: PyTorch
+    .. tab-item:: PyTorch
+        :sync: torch
 
-       .. include:: ../apiref/torch/adaround.rst
+        .. include:: ../apiref/torch/adaround.rst
+            :start-after: _apiref-torch-adaround:
```
Comment (quic-kyunggeu) [inline Docs/beta/featureguide/adaround.rst:208]: 👍

File: Docs/beta/featureguide/adaround.rst [hunk 4]
```diff
@@ -152,40 +168,53 @@ Next steps
...
 
 If the model is still not accurate enough, the next step is typically to try :ref:`quantization-aware training <featureguide-qat>`.
 
 
 API
 ===
 
-.. tabs::
+.. tab-set::
+    :sync-group: platform
```
Comment (quic-kyunggeu) [inline Docs/beta/featureguide/adaround.rst:202]: 👍

File: Docs/beta/snippets/torch/apply_adaround.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # =============================================================================
 #  @@-COPYRIGHT-START-@@
 #
-#  Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+#  Copyright (c) 2023, Qualcomm Innovation Center, Inc. All rights reserved.
```
Comment (quic-kyunggeu) [inline Docs/beta/snippets/torch/apply_adaround.py:5]: minor: We should revert this change and keep it as 2024.
We can handle it in a separate PR

File: Docs/beta/snippets/torch/evaluate.py [hunk 6]
```diff
@@ -34,7 +34,8 @@
...
 # =============================================================================
-# pylint: disable=all
-# Determine simulated accuracy
-accuracy = ImageNetDataPipeline.evaluate(sim.model, use_cuda)
-print(accuracy)
+# pylint: disable=missing-docstring
+
+    # Determine simulated accuracy
+    accuracy = ImageNetDataPipeline.evaluate(sim.model, use_cuda)
+    print(accuracy)
```
Comment (quic-kyunggeu) [inline Docs/beta/snippets/torch/evaluate.py]: Was there a reason for this indentation?
This makes the indentation levels of Step2 and Step3 inconsistent, right?
image

File: Docs/beta/snippets/torch/export.py [hunk 7]
```diff
@@ -34,8 +34,9 @@
...
-# Export the model
-# Export the model which saves pytorch model without any simulation nodes and saves encodings file for both
-# activations and parameters in JSON format
-model.export(path='./', filename_prefix='<name_prefix>', dummy_input=dummy_input.cpu())
+# pylint: disable=missing-docstring
+
+    # Export the model
+    # Export the model which saves pytorch model without any simulation nodes and saves encodings file for both
+    # activations and parameters in JSON format
+    model.export(path='./', filename_prefix='<name_prefix>', dummy_input=dummy_input.cpu())
```
Comment (quic-kyunggeu) [inline Docs/beta/snippets/torch/export.py]: Same comment as above regarding the indentation

File: Docs/beta/conf.py [hunk 8]
```diff
@@ -167,7 +176,16 @@ def setup(app):
...
 # 'searchbox.html']``.
 #
-html_sidebars = {'**': ['globaltoc.html', 'localtoc.html', 'searchbox.html'] }
+# furo theme sidebars
+html_sidebars = {
+    "**": [
+        "sidebar/brand.html",
+        "sidebar/scroll-start.html",
+        "sidebar/navigation.html",
+#        "sidebar/search.html",
```
Comment (quic-kyunggeu) [inline Docs/beta/conf.py:188]: Was this commented out intentionally? Seems like you do have the css file for search.html, don't you?

Comment (dwelsch-esi) [reply Docs/beta/conf.py:188]: Yes. It didn't seem to work but it's a low priority right now. I left it there to work on later.

File: Docs/beta/snippets/torch/evaluate.py (hunk 6 above)
Comment (dwelsch-esi) [reply Docs/beta/snippets/torch/evaluate.py]: Probably just a cut/paste error when I copied the function. Fixed.

File: Docs/beta/snippets/torch/export.py (hunk 7 above)
Comment (dwelsch-esi) [reply Docs/beta/snippets/torch/export.py]: Same.

**Review by quic-kyunggeu — APPROVED** [review]

**Review by quic-akhobare — APPROVED** [review]
